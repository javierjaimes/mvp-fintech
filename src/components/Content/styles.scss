.container {

  .table {
    display: flex;
    flex-direction: column;
    flex-wrap: nowrap;
    justify-content: flex-start;
    align-content: stretch;
    align-items: flex-start;
  }
  .row {
    display: flex;
    flex-direction: row;
    flex-wrap: nowrap;
    justify-content: flex-start;
    align-content: stretch;
    align-items: flex-start;
    border-bottom: solid 1px lightgray;
    min-width: 37rem;
    
    .field {
      margin: 1rem;
      min-width: 5rem;

      &.head{
        font-size: 14px;
        color: #BC2053;
        text-transform: uppercase;
        font-weight: bold;
      }
    }
  }
}