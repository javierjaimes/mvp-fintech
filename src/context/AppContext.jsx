import { useState, createContext } from 'react';
import MVP from '../api'

export const AppContext = createContext();

export const AppProvider = props => {
  const [list, setList] = useState([])
  return (
    <AppContext.Provider value={{list, setList, MVP}} >
      {props.children}
    </AppContext.Provider>
  )
}