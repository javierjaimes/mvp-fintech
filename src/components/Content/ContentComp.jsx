import React from 'react'
import Sidebar from '../Sidebar'
import './styles.scss'

export default function ContentComp({list, fetched, checkNRS, checkJRS, checkScore}) {
  const checkHandler = (nin) => {

  }

  const content = fetched ? (
    <div className="table">
      <div className="row">
          <div className="field head">status</div>
          <div className="field head">firstname</div>
          <div className="field head">lastname</div>
          <div className="field head">phone</div>
          <div className="field head"></div>
        </div>
      {list.map(user => (
        <div className="row" key={user.id}>
          <div className="field"><button onClick={() => checkHandler(user.nin)}>Check</button></div>
          <div className="field">{user.firstname}</div>
          <div className="field">{user.lastname}</div>
          <div className="field">{user.phone}</div>
          <div className="field">options</div>
        </div>
      ))}
    </div>
    ):(
    <div>Loading</div>
    )

  return (
    <div className="container">
      {content}
      <Sidebar />
    </div>
  )
}
