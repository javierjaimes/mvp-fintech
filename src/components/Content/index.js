import React, { useState, useContext, useEffect } from 'react'
import { AppContext } from '../../context/AppContext'
import ContentComp from './ContentComp'

export default function Content(){
  const { list, setList, MVP } = useContext(AppContext)
  const [fetched, setFetched] = useState(false)

  useEffect(() => {
    const updateData = async () => {
      const data = await MVP.getList()
      setList(data)
      setFetched(true)
    }
    !fetched && updateData()
  }, [list, setList, MVP, fetched])

  const checkNRS = async (nin) => {
    const data = await MVP.checkNRS(nin)
    return data.length > 0
  }

  const checkJRS = async (nin) => {
    const data = await MVP.checkJRS(nin)
    return data.length > 0
  }

  const checkScore = () => {
    return Math.random(0, 100)
  }

  return <ContentComp {...{list, fetched, checkNRS, checkJRS, checkScore}}  />
}