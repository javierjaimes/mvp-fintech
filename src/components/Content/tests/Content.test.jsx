import React from 'react';
import { shallow } from 'enzyme';
import Content from '../';
import { AppContext } from '../../../context/AppContext';

const list = []
const setList = jest.fn()

describe('Content', () => {
  const component = shallow(
    <AppContext.Provider value={{ list, setList }}>
      <Content />
    </AppContext.Provider>
  );

  it('should render <contentComp />', () => {
    expect(component).toMatchSnapshot();
  });
});