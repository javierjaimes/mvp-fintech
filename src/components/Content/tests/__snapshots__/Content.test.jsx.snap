// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content should render <contentComp /> 1`] = `ShallowWrapper {}`;
